src/creature.h

```diff
@@ -750,6 +752,9 @@ class Creature
         virtual void process_one_effect( effect &e, bool is_new ) = 0;
 
         pimpl<effects_map> effects;
+
+        std::map< bodypart_id, std::map<damage_type, std::pair<time_duration, int> > >damage_over_time_map;
```

Comment:
I would suggest making a class wrapper for this. it will help with maintenance and ease of understanding the code.